### ISSUE TITLE
Script that allows easier installation on arch linux based distros

### ISSUE THREAD

**Issue by JudgeMegapolis** [rapid7/metasploit-framework#16950]
## Summary

Brief explanation of the feature.

On arch linux the metasploit-framework package is outdated and don't install properly. Requires installing RVM and manually installing the gems in the directory after cloning the repository. I would like a script to automate this process or possibly some sort of repo for arch linux like with installing on debian based linux distros. I also haven't been able to find a script or even forums properly explaining how to install on arch without it ending up being broken. I had to manually look up the errors and got a vague answer on forum to install RVM first and go from there.

### Basic example/general idea

This can be easily done with bash. Install RVM and the latest version of Ruby, clone the repo or add the necessary repo to pacman. Create the necessary symlinks, environmental variables, etc. I can write this script and make a pull request if people are willing/want to add the feature. I'm not going to waste time if people aren't willing/want to add the feature.

### Motivation

Why are we doing this? What use cases does it support? What is the expected outcome?

The msf package in the repos are either very outdated or do not properly install ruby and the gems. The outcome is to just make installing the framework a bit less painful. I am aware that arch linux is for more technical people, but automating this process I think would help everyone.

**Comment by bcoles** [comment]
tl/dr: I suggest teaming up with the package maintainers.

> On arch linux the metasploit-framework package is outdated and don't install properly

Metasploit Framework packages for Arch are not managed by the Metasploit team.

If the packages are broken then the package maintainer would be the best person to contact. You can find the package maintainers here:

https://archlinux.org/packages/community/x86_64/metasploit/

Maintaining packages is a tedious, tiresome, never-ending battle and a largely thankless task. I'm sure they would appreciate assistance if you volunteered to step up and take on the responsibility.

The Metasploit team manages software installers. Does the installer work on your Arch system?

https://docs.metasploit.com/docs/using-metasploit/getting-started/nightly-installers.html

If the installer does not work then there may be interest in resolving the issues. However, support for Arch is a low priority. I doubt that Rapid7 would be interested in maintaining support Arch. The community would need to step up to this task.

> This can be easily done with bash.

Until it breaks. I'm guessing that the Metasploit package for Arch used to work once too. But now it is broken (allegedly) due to lack of community support to continue maintenance.

> I had to manually look up the errors and got a vague answer on forum to install RVM first and go from there.

Sounds like par for the course with Arch. When you install Arch, you take on the burden of managing your system. Convenience is a cost of freedom.

As per the Arch Principles on [User Centrality](https://wiki.archlinux.org/title/Arch_Linux#User_centrality):

> The distribution is intended to fill the needs of those contributing to it, rather than trying to appeal to as many users as possible. It is targeted at the proficient GNU/Linux user, or anyone with a do-it-yourself attitude who is willing to read the documentation, and solve their own problems.

**Comment by sempervictus** [comment]
I've been in touch with the Arch Ruby packagers - they're rehashing 3.1 due to stdlib dependency problems in the way the 3.0 and prior packages are built. Once that's resolved, we should be able to build the PKGBUILD correctly again.

**Comment by bcoles** [comment]
Closing in favor of Arch packages.
